apps/filebrowser/src/filebrowser/templates/listdir.mako

```diff
@@ -164,20 +164,27 @@ ${ fb_components.menubar() }
         <div class="btn-toolbar" style="display: inline; vertical-align: middle">
           % if show_upload_button:
           <!-- ko if: isS3 -->
-            <a class="btn fileToolbarBtn" title="${_('Upload files')}" data-bind="visible: !inTrash(), css: {'disabled': isS3Root()}, click: function(){ if (!isS3Root()) { uploadFile() }}"><i class="fa fa-arrow-circle-o-up"></i> ${_('Upload')}</a>
+            <a class="btn fileToolbarBtn" title="${_('Upload files')}" data-bind="visible: !inTrash(), css: {'disabled': isS3Root()}, click: function(){ if (!isS3Root()) { uploadFile(false) }}"><i class="fa fa-arrow-circle-o-up"></i> ${_('Upload')}</a>
+            <a class="btn fileToolbarBtn" title="${_('Schedule Upload')}" data-bind="visible: !inTrash(), css: {'disabled': isS3Root()}, click: function(){ if (!isS3Root()) { uploadFile(true) }}"><i class="fa fa-arrow-circle-o-up"></i> ${_('Schedule Upload')}</a>
```

Comment:
Added a check to hide schedule upload button based on "isTaskServerEnabled" flag. 

Upload button: When a file is uploaded, it initially gets written on to the local machine and the its upload to the File System like S3, Azure. Until the upload is completed on the Filesystem, the Upload pop up wont be closed. This is similar to our original implementation of Uploads in File browser. 

Schedule Upload button: When a file is uploaded, it initially gets written on to the local machine. Once its written on to the local machine, the upload pop up is closed and a Info message is displayed, "File upload scheduled. Please check the task server page for progress.". The file upload to File Systems like S3, Azure will will happen in the background. 

The advantage of schedule upload button is, when a user uploads a large file they need not wait in upload popup screen until its completed.

Reply:
I think it might be confusing to our users, what are the reasons for keeping both around? The alternative could be to have the schedule type upload only when task server is enabled and if it's disabled we use the old way.

Reply:
Yeah that makes sense. We wanted to keep both the 'upload' and 'schedule upload' buttons as per our initial plan, since the user might not be fully aware of the functionality of the 'schedule upload' button. So can we have both the buttons for now and remove it later based on user feedback? or shall we just keep one button "schedule upload"

Reply:
Sorry, my page was not showing your latest comments. A single upload alternative seems good.

Reply:
Only one button will be visible - schedule upload, when task server is enabled. Added a tool tip to display what the button does.